packages/trace-viewer/src/ui/codegen.ts

```diff
@@ -0,0 +1,305 @@
+  }
+
+  private stringLiteral(v: string): string {
+    v = v.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
+    return `"${v}"`;
+  }
+}
+
+class CSharpCodeGen implements APIRequestCodegen {
+  generatePlaywrightRequestCall(request: har.Request, body: string | undefined): string {
+    const url = new URL(request.url);
+    const urlParam = `${url.origin}${url.pathname}`;
+    const options: any = {};
+
+    const initLines: string[] = [];
+
+    let method = request.method.toLowerCase();
+    if (!['delete', 'get', 'head', 'post', 'put', 'patch'].includes(method)) {
+      options.Method = method;
+      method = 'fetch';
+    }
+
+    if (url.searchParams.size)
+      options.Params = Object.fromEntries(url.searchParams.entries());
+    if (body)
+      options.Data = body;
+    if (request.headers.length)
+      options.Headers = Object.fromEntries(request.headers.map(header => [header.name, header.value]));
+
+    const params = [`"${urlParam}"`];
+    const hasOptions = Object.keys(options).length > 0;
+    if (hasOptions)
+      params.push(this.prettyPrintObject(options));
+
+    return `${initLines.join('\n')}${initLines.length ? '\n' : ''}await request.${this.toFunctionName(method)}(${params.join(', ')});`;
+  }
+
+  private toFunctionName(method: string): string {
+    return method[0].toUpperCase() + method.slice(1) + 'Async';
+  }
+
+  private prettyPrintObject(obj: any, indent = 2, level = 0): string {
+    // Handle null and undefined
+    if (obj === null)
+      return 'null';
+    if (obj === undefined)
+      return 'null';
+
+    // Handle primitive types
+    if (typeof obj !== 'object') {
+      if (typeof obj === 'string')
+        return this.stringLiteral(obj);
+      if (typeof obj === 'boolean')
+        return obj ? 'true' : 'false';
+      return String(obj);
+    }
+
+    // Handle arrays
+    if (Array.isArray(obj)) {
+      if (obj.length === 0)
+        return 'new object[] {}';
+      const spaces = ' '.repeat(level * indent);
+      const nextSpaces = ' '.repeat((level + 1) * indent);
+
+      const items = obj.map(item =>
+        `${nextSpaces}${this.prettyPrintObject(item, indent, level + 1)}`
+      ).join(',\n');
+
+      return `new object[] {\n${items}\n${spaces}}`;
+    }
+
+    // Handle regular objects
+    if (Object.keys(obj).length === 0)
+      return 'new {}';
+    const spaces = ' '.repeat(level * indent);
+    const nextSpaces = ' '.repeat((level + 1) * indent);
+
+    const entries = Object.entries(obj).map(([key, value]) => {
+      const formattedValue = this.prettyPrintObject(value, indent, level + 1);
+      const formattedKey = level === 0 ? key : `[${this.stringLiteral(key)}]`;
+      return `${nextSpaces}${formattedKey} = ${formattedValue}`;
+    }).join(',\n');
+
+    return `new() {\n${entries}\n${spaces}}`;
+  }
+
+  private stringLiteral(v: string): string {
+    // escape douvle quotes and backslashes
+    v = v.replace(/\\/g, '\\\\').replace(/"/g, '\\"');
```

Comment:
```suggestion
    v = v.replace(/\\/g, '\\\\').replace(/"/g, '\\"').replace(/\n/g, '\\n').replace(/\r/g, '\\r').replace(/\t/g, '\\t');
  return $"\"{v}\"";
```

Reply:
Now that I think about it, I think this one could use `JSON.stringify`!